### PR TITLE
benchmarks: add ng2_ftl benchmark

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/benchmarks/src/tree/ng2_ftl/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- forceGc: false
+- regressionSlopeMetric: scriptTime
+- runId: f9ae32f0-8580-11e6-914d-9f4f8dbfb5e8
+- sampleSize: 20
+- userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.116 Safari/537.36
+
+...createOnly               gcAmount |             gcTime |        majorGcTime |     pureScriptTime |         renderTime |         scriptTime
+=============== | ================== | ================== | ================== | ================== | ================== | ==================
+ng2.static      |      10068.17+-33% |          9.50+-43% |          0.01+-84% |         83.01+-18% |         57.81+-19% |         92.33+-16%
+ng2.ftl         |       493.36+-435% |         0.65+-435% |         0.00+-435% |         28.48+-16% |          53.97+-7% |         28.48+-16%
```
Comment (vicb) [inline modules/benchmarks/src/tree/ng2_ftl/README.md]: :+1:

File: modules/benchmarks/src/tree/ng2_ftl/index.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import {platformBrowser} from '@angular/platform-browser';
+
+import {bindAction, profile} from '../../util';
+import {TreeNode, buildTree, emptyTree} from '../util';
+
+import {AppModuleNgFactory} from './app.ngfactory';
+import {TreeRootComponent} from './tree';
+
+export function main() {
+  var tree: TreeRootComponent;
```
Comment (vicb) [inline modules/benchmarks/src/tree/ng2_ftl/index.ts]: let * 2

**Comment by chuckjaz** [comment]
@tbosch: I am seeing errors like 'Error: ENOTDIR: not a directory, stat '/Users/chuckj/src/angular/dist/packages-dist/common/bundles/common.umd.js/src/common_module.js' both locally and on Travis which look specific to this PR (that is, they  do not happen on master).

File: modules/benchmarks/src/tree/ng2_ftl/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+I.e. right now, the Ng2 compiler _does not_ produce this kind of code,
+but hopefully will in the future.
+
+## Optimizations and cleanups
+
+1. remove view factory functions, call the constructor of the view class directly
+2. remove `createInternal` and move the logic into the constructor
+3. remove `AppView` parent class
+4. generate direct calls to `detectChangesInternal` / `destroyInternal` of nested components
+5. remove `Renderer` and `SantizationService` to update
```
Comment (alfaproject) [inline modules/benchmarks/src/tree/ng2_ftl/README.md]: Typo

File: modules/benchmarks/src/tree/ng2_ftl/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# Ng2 Faster Than BaseLine (FTL) Benchmark
+
+This benchmark was produced in the following way:
+1. Run AoT over the root component, one branch component
```
Comment (alfaproject) [inline modules/benchmarks/src/tree/ng2_ftl/README.md]: Missing new line to make a ordered list

File: modules/benchmarks/src/tree/ng2_ftl/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+- bundles: false
+- depth: 11
+- forceGc: false
+- regressionSlopeMetric: scriptTime
+- runId: f9ae32f0-8580-11e6-914d-9f4f8dbfb5e8
+- sampleSize: 20
+- userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.116 Safari/537.36
+
+...createOnly               gcAmount |             gcTime |        majorGcTime |     pureScriptTime |         renderTime |         scriptTime
+=============== | ================== | ================== | ================== | ================== | ================== | ==================
```
Comment (alfaproject) [inline modules/benchmarks/src/tree/ng2_ftl/README.md]: https://help.github.com/articles/organizing-information-with-tables/

Comment (tbosch) [inline modules/benchmarks/src/tree/ng2_ftl/README.md]: Thx, I just took the output of benchpress and dumped it here. But you are right, I could make this nicer!

File: modules/benchmarks/src/tree/ng2_ftl/index.ts (hunk 2 above)
Comment (tbosch) [inline modules/benchmarks/src/tree/ng2_ftl/index.ts]: Done

**Pull request merged**